File: source/_posts/Algorithm/1.leetcode/26.删除排序数组中的重复项/26.删除排序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除排序数组中的重复项
#


# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return 0
        index = 0
        for item in nums[1:]:
            if item != nums[index]:
                index += 1
                nums[index] = item
        return index + 1


# @lc code=end

File: source/_posts/Algorithm/1.leetcode/Easy/26.删除排序数组中的重复项/26.删除排序数组中的重复项.go
/*
 * @lc app=leetcode.cn id=26 lang=golang
 *
 * [26] 删除排序数组中的重复项
 */
package main

// @lc code=start
func removeDuplicates(nums []int) int {
	n := len(nums)
	if n < 2 {
		return n
	}
	l, r := 0, 1
	for r < n {
		if nums[l] < nums[r] {
			l++
			nums[l] = nums[r]
		}
		r++
	}
	return l + 1
}

// @lc code=end
